src/boxrobots/src/controllers/QPinnerloopcontroller.jl

```diff
@@ -0,0 +1,476 @@
+
+  # Returns
+  * next_contact_mismatch
+  """
+  planned_contact_state = contact_plan.plan(t_plan)
+  nxt_contact_switch_time = get_next_contact_switch_time(contact_plan)
+
+
+  if t_plan >= nxt_contact_switch_time
+    print("\n\n-----------------------")
+    print("-----------------------")
+    print("\n\n----- t_plan > nxt_contact_switch_time --------\n\n")
+    # could be making OR breaking contact
+
+    # this should be the same as planned_contact_state in this case
+    nxt_planned_contact_state = get_next_contact_state(contact_plan)
+    has_contact_switch, contact_switch = ContactSwitch(contact_state, planned_contact_state)
+
+
+    # some debugging print statements
+    if false
+      println("t_plan = ", t_plan)
+      println("contact_state = ", contact_state)
+      println("planned_contact_state = ", planned_contact_state)
+      println("nxt_planned_contact_state = ", nxt_planned_contact_state)
+    end
+
+    if !has_contact_switch
+      print("---- made contact switch on time ----")
+      print("t_plan = ", t_plan)
+      print("\n\n")
+      # made contact in time
+      if !isequal(nxt_planned_contact_state, contact_state)
+        error("nxt_planned_contact_state not same as contact_state")
+      end
+      # increment the next_contact_state_idx counter
+      increment_next_contact_state_idx!(contact_plan)
+      return contact_mismatch
+    end
+
+    # if we got here then there is a mismatch between contact_state and
+    # and planned_contact_state
+    contact_switch_type = get_contact_switch_type(contact_switch)
+
+    if contact_switch_type == MakingContact
+      print("\n\n-----------------------")
+      print("-----------------------")
+      print("\n\n----------Late Contact----------\n")
+      print("t_plan = ", t_plan)
+      print("\n\n")
+      print("\n\n-----------------------\n\n")
+      nxt_contact_mismatch = ContactMismatch(contact_switch.limb_symbol, LateContact)
+      increment_next_contact_state_idx!(contact_plan)
+      return nxt_contact_mismatch
+    else
+      print("\n\n-----------------------")
+      print("-----------------------")
+      print("\n\n----------Breaking Contact----------\n")
+      print("t_plan = ", t_plan)
+      print("\n\n")
+      print("\n\n-----------------------\n\n")
+
+      # must be breaking contact, so we have no mismatch, just go with planned
+      # contact assignments
+      # Increment next_contact_state_idx in the contact_plan
+      increment_next_contact_state_idx!(contact_plan)
+      return contact_mismatch
+    end
+  else
+    # could have NoMismatch or EarlyContact
+    has_contact_switch = !isequal(contact_state, planned_contact_state)
+
+    if has_contact_switch
+      print("\n\n-----------------------\n\n")
+      print("\n\n----------Early Contact-----------")
+      print("t_plan = ", t_plan)
+      print("\n\n")
+      # we should be in EarlyContact here
+      ~, contact_switch = ContactSwitch(planned_contact_state, contact_state)
```

Comment:
`_` is used for discarded values by convention in Julia, unlike Matlab. In fact, future Julia versions will probably introduce some optimizations so that `_` can be explicitly used to throw away unused data